### PR TITLE
[ci] `pkg/resource/deploy/deploytest/provider.go` coverage

### PR DESCRIPTION
**Pull request by dixler**
Adds `pkg/resource/deploy/deploytest/provider_test.go` which adds missing coverage to `pkg/resource/deploy/deploytest/provider.go`

Note: disregard CI changes. will revert prior to merge. (Left solely to illustrate coverage changes).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -358,7 +358,7 @@ jobs:
           ls -lhR test-results
           find test-results -mindepth 1 -name '*.json' -mtime +7 -delete
   test-collect-coverage:
-    needs: [unit-test, integration-test, acceptance-test]
+    needs: [unit-test, integration-test]
```
Comment (tgummerer) [inline .github/workflows/ci.yml:361]: Are the workflows and `go-test.py` changes meant to be merged as well?

File: .github/workflows/on-pr.yml [hunk 2]
```diff
@@ -106,7 +106,7 @@ jobs:
           || ''
         }}
       # We'll only upload coverage artifacts with the periodic-coverage cron workflow.
```
Comment (tgummerer) [inline .github/workflows/on-pr.yml:108]: If this is meant to be merged, this comment is outdated.

File: pkg/resource/deploy/deploytest/provider.go [hunk 3]
```diff
@@ -203,7 +205,11 @@ func (prov *Provider) Construct(info plugin.ConstructInfo, typ tokens.Type, name
 	if prov.ConstructF == nil {
 		return plugin.ConstructResult{}, nil
 	}
-	monitor, err := dialMonitor(context.Background(), info.MonitorAddress)
+	dialMonitorImpl := dialMonitor
+	if prov.DialMonitorF != nil {
```
Comment (Frassle) [inline pkg/resource/deploy/deploytest/provider.go:209]: I wonder if this should actually just push this logic down into CallF/ConstructF. If we passed the address string to ConstructF/CallF and exposed DialMonitor for tests to use where that's appropriate, and for these new tests they just short circuit to only looking at the string.

File: .github/workflows/on-pr.yml (hunk 2 above)
Comment (dixler) [reply .github/workflows/on-pr.yml:108]: > Note: disregard CI changes. will revert prior to merge.

Comment (dixler) [reply .github/workflows/on-pr.yml:108]: This comment is outdated, but outside the scope of this PR.

File: pkg/resource/deploy/deploytest/provider.go (hunk 3 above)
Comment (dixler) [reply pkg/resource/deploy/deploytest/provider.go:209]: That's interesting. I'm not sure of other tests that expect `dialMonitor` will be called, but we should not change existing behavior by not calling it as this PR is focused on adding test coverage without modifying behavior.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dixler) [reply .github/workflows/ci.yml:361]: No. They're to illustrate coverage changes.

File: pkg/resource/deploy/deploytest/provider.go (hunk 3 above)
Comment (Frassle) [reply pkg/resource/deploy/deploytest/provider.go:209]: deploytest is literally just for our tests, it's fine to refactor

Comment (justinvp) [reply pkg/resource/deploy/deploytest/provider.go:209]: > I wonder if this should actually just push this logic down into CallF/ConstructF. If we passed the address string to ConstructF/CallF and exposed DialMonitor for tests to use where that's appropriate, and for these new tests they just short circuit to only looking at the string.

Seems like a reasonable suggestion! However, I'm not sure if we can do this refactor and still hit our coverage goal by EOW. I suggest we go with `DialMonitorF` for now to continue to make progress, and follow-up with potential refactor after.

File: pkg/resource/deploy/deploytest/provider_test.go [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+	t.Parallel()
+	t.Run("SignalCancellation", func(t *testing.T) {
+		t.Parallel()
+		t.Run("has CancelF", func(t *testing.T) {
+			t.Parallel()
+			var called bool
+			prov := &Provider{
+				CancelF: func() error {
+					called = true
+					return fmt.Errorf("expected error")
```
Comment (justinvp) [inline pkg/resource/deploy/deploytest/provider_test.go:39]: Nit: this should be `errors.New`

**Review by justinvp — APPROVED** [review]
Looks reasonable to me

File: pkg/resource/deploy/deploytest/provider_test.go (hunk 4 above)
Comment (Frassle) [reply pkg/resource/deploy/deploytest/provider_test.go:39]: I thought I turned a linter on for this!?

Comment (Frassle) [reply pkg/resource/deploy/deploytest/provider_test.go:39]: This is what perfsprint is supposed to pick up? Is this file not being linted at all?

**Pull request merged**